src/sql/src/plan/statement/dml.rs

```diff
@@ -160,25 +166,128 @@ pub fn plan_read_then_write(
     }))
 }
 
+fn is_side_effecting(
+    scx: &StatementContext,
+    select: &SelectStatement<Aug>,
+) -> Result<Option<SideEffectingFunction>, PlanError> {
```

Comment:
```suggestion
/// Plans the `SELECT` if it contains calls to side-effecting functions.
///
/// In PostgreSQL, these functions can appear anywhere in a query:
///
///     SELECT 1 WHERE pg_cancel_backend(1234)
///
/// In Materialize, our compute layer cannot execute functions with side effects. So we
/// sniff out the common form of function calls, i.e.
///
///    SELECT side_effecting_function(...)
///
/// where all arguments are literals or bound parameters, and plan them specially as a
/// `Plan::SideEffectingFunction`. This gets us compatibility with PostgreSQL for 90% of
/// real-world use cases, without causing stress for the compute layer (optimizer,
/// dataflow execution, etc.), as we can handle all the side effects cleanly in the adapter layer.
fn plan_select_if_side_effecting(
    // This match looks unwieldy, but it should be robust in practice. We exhaustively
    // match everywhere, so the compiler will warn us if we add new fields to the AST
    // that need to be handled here. 
```

(The `is_` naming makes me think it's just checking, when in fact it is also planning if it's a single side-ffecting function.)